### DISCUSSION TITLE
Best method to extract an entire archive?

### DISCUSSION THREAD

**Discussion by gbook** [rikyoz/bit7z#263] [General]
What is the fastest/best method to extract an entire archive to a specified directory? I see there are extract() functions for the BitArchiveReader and BitArchiveExtractor classes.

**Comment by rikyoz** [comment]
> What is the fastest/best method to extract an entire archive to a specified directory?

Both [`BitArchiveReader::extractTo`](https://github.com/rikyoz/bit7z/wiki/BitArchiveReader#void-extractto-const-tstring-outdir--const) and [`BitFileExtractor::extract`](https://github.com/rikyoz/bit7z/wiki/BitFileExtractor#void-extract-const-tstring-inarchive-const-tstring-outdir----const) should perform the same, as the underlying code is mostly the same.

In general, the `BitFileExtractor` class is designed for the extraction of multiple archive files with the same set of callback functions and settings. In contrast, the `BitArchiveReader` class is intended for the extraction of only a single archive file.
